### PR TITLE
Assign correct span to continue expression

### PR DESCRIPTION
**Pull request by marcusklaas**
Fixes https://github.com/rust-lang/rust/issues/28105.

### CODE REVIEW

File: src/test/compile-fail/issue-28105.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Make sure that a continue span actually contains the keyword.
+
+fn main() {
+    'a: loop {
+        if false {
+            continue //~ ERROR use of undeclared label
```
Comment (nagisa) [inline src/test/compile-fail/issue-28105.rs]: This test will get broken as soon as the label thing gets fixed.

How about 

```rust
fn main() {
    continue //~ ERROR `continue` outside of loop
    ;
    break //~ ERROR `break` outside of loop
    ;
}
```

instead?

Comment (marcusklaas) [inline src/test/compile-fail/issue-28105.rs]: That's a lot better. Can we still change this? @bors seems to be merging already..

Comment (nagisa) [inline src/test/compile-fail/issue-28105.rs]: If this doesn’t bounce, you can change it as a follow-up PR.

Comment (marcusklaas) [inline src/test/compile-fail/issue-28105.rs]: Cool. I'm making a new PR for a related issue any way.

**Pull request merged**